src/profilo.php

```diff
@@ -121,9 +138,14 @@ function checkMoney($money, &$messaggi) {
         if ($_SERVER['REQUEST_METHOD'] === 'POST' && isset($_POST['aggiungi-saldo'])){
             $saldo = $database->pulisciInput($_POST['saldo']);
             if (checkMoney($saldo, $avvisoSaldo)) {
-                $query = "UPDATE utente SET saldo = saldo + ? WHERE username = ?";
-                $values = [$saldo, $username];
-                $avvisoSaldo .= $database->executeCRUDPreparedStatement($query, 'ds', $values);
+                if (!isSaldoOverflow($database, $username, $saldo)) {
+                    $query = "UPDATE utente SET saldo = saldo + ? WHERE username = ?";
+                    $values = [$saldo, $username];
+                    $avvisoSaldo .= $database->executeCRUDPreparedStatement($query, 'ds', $values);
+                }
+                else {
+                    $avvisoSaldo .= makeMessageParagraph("Questa aggiunta potrebbe sforare il tetto massimo del saldo possedibile!");
```

Comment:
```suggestion
                    $avvisoSaldo .= makeMessageParagraph("Questa aggiunta potrebbe sforare il tetto massimo del saldo possedibile, che sarebbe 99999.99999!");
```
aggiungerei anche la cifra per non mettere in confusione e dare un suggerimento